inventory/rego/policy.rego

```diff
@@ -78,18 +47,51 @@ deny if {
 	input.ClientKind == "CLIENT_KIND_API"
 }
 
-allow if {
-	not input.DesiredState
+# deny if a tenant resource is created/updated by other than TC client
+deny if {
+	input.resource.tenant
+	not input.ClientKind == "CLIENT_KIND_TENANT_CONTROLLER"
+}
+
+# deny updates to the CurrentState via northbound API
+deny if {
+	input.CurrentState
+	input.ClientKind == "CLIENT_KIND_API"
+}
+
+# deny updates to the DesiredState via southbound API
+deny if {
+	input.DesiredState
+	input.ClientKind == "CLIENT_KIND_RESOURCE_MANAGER"
+}
+
+# deny updates to the currentPowerState via northbound API
+deny if {
 	input.resource.host
 	input.resource.host.currentPowerState
-	not input.resource.host.desiredPowerState
+	input.ClientKind == "CLIENT_KIND_API"
+}
+
+# deny updates to the desiredPowerState via southbound API
+deny if {
+	input.resource.host
+	input.resource.host.desiredPowerState
 	input.ClientKind == "CLIENT_KIND_RESOURCE_MANAGER"
 }
 
+# deny if TC client tries to create a resource that is not a tenant, provider or telemetryGroup
```

Comment:
I'm not entirely sure if we want to keep this rule. This limits a lot the evolution of TC. Anyway, we can keep it for now.